### PR TITLE
[docs] Update old forums link to Discord forums

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Remove old forums links and replace them by discord forums & closes ENG-10240.

~~Also, updated the forums icon on the home page to use the same icon from the footer and sidebar footer.~~

**Update**:

After talking with Kim and Simek, we decided to use a link on the home page and footer sidebar for Discord and Forums.
CleanShot 2023-10-18 at 15 49 33@2x

CleanShot 2023-10-17 at 03 57 25@2x

# Test Plan

Run docs locally, visit a docs page and click the "Ask a question in the forums..." link from the footer of that page. It should redirect to Discord.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by kbrandwijk — APPROVED** [review]

**Review by kbrandwijk — APPROVED** [review]

File: docs/pages/debugging/errors-and-warnings.mdx [hunk 1]
```diff
@@ -31,7 +31,7 @@ This stack trace is **extremely valuable** since it gives you the location of th
...
 
-Debugging errors is one of the most frustrating but satisfying parts of development. Remember that you're never alone. The **Expo community** and the React and React Native communities are great resources for help when you get stuck. There's a good chance someone else has run into your exact error. Make sure to read the documentation, search the [forums](https://forums.expo.dev/), [GitHub issues](https://github.com/expo/expo/issues/), and [Stack Overflow](https://stackoverflow.com/).
+Debugging errors is one of the most frustrating but satisfying parts of development. Remember that you're never alone. The **Expo community** and the React and React Native communities are great resources for help when you get stuck. There's a good chance someone else has run into your exact error. Make sure to read the documentation, search the [forums](https://discord.gg/expo), [GitHub issues](https://github.com/expo/expo/issues/), and [Stack Overflow](https://stackoverflow.com/).
```
Comment (kbrandwijk) [inline docs/pages/debugging/errors-and-warnings.mdx:34]: Why would we use the discord.gg link here vs chat.expo.dev?

Comment (amandeepmittal) [inline docs/pages/debugging/errors-and-warnings.mdx:34]:
```suggestion
Debugging errors is one of the most frustrating but satisfying parts of development. Remember that you're never alone. The **Expo community** and the React and React Native communities are great resources for help when you get stuck. There's a good chance someone else has run into your exact error. Make sure to read the documentation, search the [forums](https://chat.expo.dev/), [GitHub issues](https://github.com/expo/expo/issues/), and [Stack Overflow](https://stackoverflow.com/).
```

File: docs/pages/debugging/runtime-issues.mdx [hunk 2]
```diff
@@ -171,7 +171,7 @@ This might indicate that there is a performance issue. You likely need to run yo
 
 ## Stuck?
 
-The Expo community and the React and React Native communities are great resources for help when you get stuck. There's a good chance someone else has run into the same error as you, so make sure to read the documentation, search the [forums](https://forums.expo.dev/), [GitHub issues](https://github.com/expo/expo/issues/), and [Stack Overflow](https://stackoverflow.com/).
+The Expo community and the React and React Native communities are great resources for help when you get stuck. There's a good chance someone else has run into the same error as you, so make sure to read the documentation, search the [forums](https://discord.gg/expo), [GitHub issues](https://github.com/expo/expo/issues/), and [Stack Overflow](https://stackoverflow.com/).
```
Comment (amandeepmittal) [inline docs/pages/debugging/runtime-issues.mdx:174]:
```suggestion
The Expo community and the React and React Native communities are great resources for help when you get stuck. There's a good chance someone else has run into the same error as you, so make sure to read the documentation, search the [forums](https://chat.expo.dev/), [GitHub issues](https://github.com/expo/expo/issues/), and [Stack Overflow](https://stackoverflow.com/).
```

File: docs/pages/debugging/errors-and-warnings.mdx (hunk 1 above)
Comment (kbrandwijk) [reply docs/pages/debugging/errors-and-warnings.mdx:34]: Additional benefit is that we can track chat.expo.dev, but not the discord.gg one directly.

File: docs/pages/build-reference/troubleshooting.mdx [hunk 3]
```diff
@@ -61,7 +61,7 @@ It's important to note that with iOS builds the build details page only displays
 
 If you are working on a managed app and the build error is a native error rather than a JavaScript error, this is likely due to a [config plugin](/config-plugins/introduction/) or a dependency in your project. Keep an eye out in the logs for any new packages that you've added since your previous successful build. Run `npx expo-doctor` to determine that the versions of Expo SDK dependencies in your project are compatible with your Expo SDK version.
 
-Armed with your error logs, you can often start to fix your build, or you can search the [forums](https://forums.expo.dev) and GitHub issues for related packages to dig deeper. Some common sources of problems are listed below.
+Armed with your error logs, you can often start to fix your build, or you can search the [forums](https://discord.gg/expo) and GitHub issues for related packages to dig deeper. Some common sources of problems are listed below.
```
Comment (amandeepmittal) [inline docs/pages/build-reference/troubleshooting.mdx:64]:
```suggestion
Armed with your error logs, you can often start to fix your build or search the [forums](https://chat.expo.dev/) and GitHub issues for related packages to dig deeper. Some common sources of problems are listed below.
```

File: docs/pages/workflow/ios-simulator.mdx [hunk 4]
```diff
@@ -38,7 +38,7 @@ Run your app with `npx expo start` and press <kbd>i</kbd> from the command line.
 
 You may get a warning about needing to accept the Xcode license. Run the command that it suggests. Open your app again to see if it was successful. If not, check the [troubleshooting](#troubleshooting) tips below.
 
-If the troubleshooting tips are not helpful, seek help on [Expo Development Tools section of the forums](https://forums.expo.dev/c/expo-dev-tools), Stack Overflow, or Google.
+If the troubleshooting tips are not helpful, ask a question in our [forums](https://discord.gg/expo), Stack Overflow, or Google.
```
Comment (amandeepmittal) [inline docs/pages/workflow/ios-simulator.mdx:41]:
```suggestion
If the troubleshooting tips are not helpful, ask a question in our [forums](https://chat.expo.dev/), Stack Overflow, or Google.
```

File: docs/ui/components/Footer/Links.tsx [hunk 5]
```diff
@@ -24,18 +24,14 @@ export const IssuesLink = ({ title, repositoryUrl }: { title: string; repository
 export const ForumsLink = ({ isAPIPage, title }: { isAPIPage: boolean; title: string }) =>
   isAPIPage ? (
     <LI>
-      <A
-        isStyled
-        openInNewTab
-        href={`https://forums.expo.dev/tag/${title}`}
-        className={LINK_CLASSES}>
+      <A isStyled openInNewTab href="https://discord.gg/expo" className={LINK_CLASSES}>
```
Comment (amandeepmittal) [inline docs/ui/components/Footer/Links.tsx:27]:
```suggestion
      <A isStyled openInNewTab href="https://chat.expo.dev/" className={LINK_CLASSES}>
```

File: docs/ui/components/Footer/Links.tsx [hunk 6]
```diff
@@ -24,18 +24,14 @@ export const IssuesLink = ({ title, repositoryUrl }: { title: string; repository
...
         <MessageDotsSquareIcon className={ICON_CLASSES} />
         <CALLOUT theme="secondary">Ask a question on the forums about {title}</CALLOUT>
       </A>
     </LI>
   ) : (
     <LI>
-      <A isStyled openInNewTab href="https://forums.expo.dev/" className={LINK_CLASSES}>
+      <A isStyled openInNewTab href="https://discord.gg/expo" className={LINK_CLASSES}>
```
Comment (amandeepmittal) [inline docs/ui/components/Footer/Links.tsx:34]:
```suggestion
      <A isStyled openInNewTab href="https://chat.expo.dev/" className={LINK_CLASSES}>
```

File: docs/pages/faq.mdx [hunk 7]
```diff
@@ -11,7 +11,7 @@ import {
   BellRinging04Icon,
 } from '@expo/styleguide-icons';
 
-This page lists some of the common questions and answers about Expo and related services. If you have a question that is not answered here, see [ Forums](https://forums.expo.io) for more common questions.
+This page lists some of the common questions and answers about Expo and related services. If you have a question that is not answered here, see [Forums](https://discord.gg/expo) for more common questions.
```
Comment (amandeepmittal) [inline docs/pages/faq.mdx:14]:
```suggestion
This page lists some of the common questions and answers about Expo and related services. If you have a question that is not answered here, see [Forums](https://chat.expo.dev/) for more common questions.
```

File: docs/pages/debugging/errors-and-warnings.mdx (hunk 1 above)
Comment (amandeepmittal) [reply docs/pages/debugging/errors-and-warnings.mdx]: All links fixed.

File: docs/pages/index.tsx [hunk 8]
```diff
@@ -275,8 +274,8 @@ export function JoinTheCommunity() {
             icon={<GithubIcon className="icon-lg text-palette-white" />}
           />
           <CommunityGridCell
-            title="Discord"
-            description="Join our Discord and chat with other Expo users."
+            title="Discord and Forums"
+            description="Join our Discord to chat with Expo users or ask questions on the forums."
```
Comment (amandeepmittal) [inline docs/pages/index.tsx:278]:
```suggestion
            description="Join our Discord to chat with Expo users or ask questions."
```

File: docs/pages/build-reference/troubleshooting.mdx [hunk 9]
```diff
@@ -154,7 +154,7 @@ If you have followed the advice here, you're now in a good position to describe
 
 ### How to ask a good question
 
-Join us on [Discord](https://chat.expo.dev) or the [forums](https://forums.expo.dev) to ask for help from the community and the Expo team. The Expo team does our best to respond to high quality and well articulated questions and issues, but responses are not guaranteed unless you are signed up for a [support plan](https://expo.dev/eas/enterprise-support). To ensure that an Expo team member sees your question, you can file a ticket at [expo.dev/contact](https://expo.dev/contact).
+Join us on [Discord and Forums](https://chat.expo.dev) to ask for help from the community and the Expo team. The Expo team does our best to respond to high quality and well-articulated questions and issues, but responses are not guaranteed unless you are signed up for a [support plan](https://expo.dev/eas/enterprise-support). To ensure that an Expo team member sees your question, you can file a ticket at [expo.dev/contact](https://expo.dev/contact).
```
Comment (kbrandwijk) [inline docs/pages/build-reference/troubleshooting.mdx:157]: This is not in scope for this PR, but this link to Enterprise Support is a bit off. Responses are guaranteed on any paid plan, this should link to our response times section imho https://expo.dev/support-terms#target-response-time-guidelines-for-subscriptions

Comment (amandeepmittal) [inline docs/pages/build-reference/troubleshooting.mdx:157]:
```suggestion
Join us on [Discord and Forums](https://chat.expo.dev) to ask for help from the community and the Expo team. The Expo team does our best to respond to high quality and well-articulated questions and issues, but responses are not guaranteed unless you are signed up for a [support plan](https://expo.dev/support-terms#target-response-time-guidelines-for-subscriptions). To ensure that an Expo team member sees your question, you can file a ticket at [expo.dev/contact](https://expo.dev/contact).
```

Comment (amandeepmittal) [reply docs/pages/build-reference/troubleshooting.mdx]: TIL! We've that on our website. I am not sure when the last time this doc was updated or audited but I've updated it with the link you shared.

**Review by kbrandwijk — APPROVED** [review]
Thank you for your conscientious updates!

**Review by Simek — APPROVED** [review]
Thank you for all of the updates! LGTM 👍

**Pull request merged**